music_service_async_interface/__init__.py

```diff
@@ -11,19 +12,63 @@
     AsyncSeekableHTTPFile = None
 
 
-# TODO [#5]: Specify how to resolve collections of collections of tracks
+# TODO [#5]: Specify how to implement collections of collections of tracks
 #   eg. user containing playlists containing tracks or artist containing albums containing tracks
 
-# TODO [#6]: AudioQuality abstract class extending enum with comparability by definition order
-#   https://docs.python.org/3/library/enum.html#using-a-custom-new
-#   https://docs.python.org/3/library/enum.html#orderedenum
-#   hold definition order as additional prop
-
 
 class InvalidURL(Exception):
     pass
 
 
+class AudioQuality(enum.Enum):
+    """Comparable enum for definition of track's audio quality.
+
+    Qualities are sorted in order of definition.
+    First defined is lower quality than second defined.
+
+    Simple example:
+    >>> class TestAudioQuality(AudioQuality):
+    >>>     LOW = 'low'
+    >>>     MEDIUM = 'med'
+    >>>     HIGH = 'high'
+    >>>
+    >>> TestAudioQuality.LOW > TestAudioQuality.HIGH
+    False
+    >>> TestAudioQuality.LOW >= TestAudioQuality.HIGH
+    False
+    >>> TestAudioQuality.LOW < TestAudioQuality.HIGH
+    True
+    >>> TestAudioQuality.LOW <= TestAudioQuality.HIGH
+    True
+    >>> TestAudioQuality.MEDIUM > TestAudioQuality.LOW
+    True
+    >>> TestAudioQuality.MEDIUM > TestAudioQuality.HIGH
+    False
+    >>> TestAudioQuality.MEDIUM < TestAudioQuality.HIGH
+    False
+    """
+
+    def __ge__(self, other):
+        if self.__class__ is other.__class__:
```

Comment:
Why not `isinstance`?

Reply:
Because to be comparable it needs to be exactly the same class, not subclass.
eg. You shouldn't be able to compare TidalAudioQuality with DeezerAudioQuality